Added block device facts

bcoca opened the pull request:
Setup now adds block device facts, (linux only for now), mostly copied from:
  https://raw.github.com/kavink/ansible/devel/library/get_devices

tested on debian and gentoo, will probably followup with freeBSD version soon

bcoca committed:
added block device info gathering
Signed-off-by: Brian Coca <briancoca+dev@gmail.com>

bcoca committed:
corrected partition sector info, added start also
Signed-off-by: Brian Coca <briancoca+dev@gmail.com>

bcoca committed:
added prettyfing byte function
Signed-off-by: Brian Coca <briancoca+dev@gmail.com>

bcoca committed:
now disk also shows pretty byte formatting
Signed-off-by: Brian Coca <briancoca+dev@gmail.com>

bcoca committed:
limits now properly set for high side
Signed-off-by: Brian Coca <briancoca+dev@gmail.com>

bcoca committed:
refactored code, now uses std function in module
updated comments
Signed-off-by: Brian Coca <briancoca+dev@gmail.com>

bcoca committed:
fixed missing sysdir to get some of the metadata
Signed-off-by: Brian Coca <briancoca+dev@gmail.com>

dhozac commented:
This should be squashed before it can be merged.

You should use subprocess.Popen().communicate() rather than wait and then reading from the fd, as you can get into a hang that way depending on OS buffering.

mounts should likely just be a list, you can have the same basename or even the same device mounted in multiple places.

I'm not sure what sysfs_no_links is for, but it seems to be set at the wrong level, as previous devices would influence the results of later ones. It also looks like it should just be a bool.

The "No" and "Yes" values for removable should just be a bool.

SDD probably should be SSD. I'm not sure you can really conclude that from something not being rotational though, considering e.g. thumb drives and CF cards.

```
part['start'] = part['start'] if part['start'] else 0
```

really isn't very readable, and I wouldn't be surprised if it didn't work on Python 2.4.

bcoca commented:
Thanks for the feedback.

This should be squashed before it can be merged.

>  I'll squash once I'm happy with the state of the code, I was mostly
> putting it out there early to get people looking at it sooner rather than
> later.
> 
> You should use subprocess.Popen().communicate() rather than wait and then
> reading from the fd, as you can get into a hang that way depending on OS
> buffering.
> 
> Done, hadn't realize this, good catch

mounts should likely just be a list, you can have the same basename or even

> the same device mounted in multiple places.
> 
> I had changed them to dictionaries when adding the FreeBSD version, ported
> back to Linux devices.
> 
> I'm not sure what sysfs_no_links is for, but it seems to be set at the
> wrong level, as previous devices would influence the results of later ones.
> It also looks like it should just be a bool.
> 
> I wasn't sure what this did, copied it from original code, i still haven't
> decided what to do, contacting original author.
> 
> The "No" and "Yes" values for removable should just be a bool.
> 
> SDD probably should be SSD. I'm not sure you can really conclude that from
> something not being rotational though, considering e.g. thumb drives and CF
> cards.
> 
> copied from the original, maybe 'Solid State' instead?
> 
> part['start'] = part['start'] if part['start'] else 0
> 
> really isn't very readable, and I wouldn't be surprised if it didn't work
> on Python 2.4.
> 
> Fixed, added default return value to get_file_content function removed ugly
> trinary assignment.
> 
>  —
> Reply to this email directly or view it on GitHubhttps://github.com/ansible/ansible/pull/1812#issuecomment-11909227.

## 

Brian Coca
Stultorum infinitus est numerus
0110000101110010011001010110111000100111011101000010000001111001011011110111010100100000011100110110110101100001011100100111010000100001
Pedo mellon a minno

mpdehaan commented:
Closing this so we don't merge it then, if you want to solicit feedback you can still refer to this by the URL.
